modules/base/src/lgoe.fz

```diff
@@ -0,0 +1,79 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature lgoe
+#
+# -----------------------------------------------------------------------
+
+# lower -- unit feature to denote the result of a comparion
+#
+public lower is
+
+# greater -- unit feature to denote the result of a comparion
+#
+public greater is
+
+# equal -- unit feature to denote the result of a comparion
+#
+public equal is
+
+
+# choice of lower greater or equal
+#
+public lgoe : choice lower greater equal, property.equatable is
+
+  # maps lgoe to a number
+  #
+  # lower => -1
+  # greater => +1
+  # equal => 0
+  #
+  public as_num =>
+    match lgoe.this
+      lower => -1
+      greater => +1
+      equal => 0
+
+
+  public fixed redef type.equality(a, b lgoe) bool =>
+    match a
+      lower =>
+        match b
+          lower => true
+          * => false
+      greater =>
+        match b
+          greater => true
+          * => false
+      equal =>
+        match b
+          equal => true
+          * => false
+
+
+
+  # human readable string of this choice
+  #
+  public redef as_string =>
+    match lgoe.this
+      lower => "lower"
+      greater => "greater"
+      equal => "equal"
+
```

Comment:
Maybe a set of predicates `is_less`, `is_less_or_equal`, etc. for all 6 non-trivial combinations would be nice here.